packages/@azure/eventhubs/client/lib/linkEntity.ts

```diff
@@ -180,6 +180,28 @@ export class LinkEntity {
     }
   }
 
+  /**
+   * Removes the Sender|Receiver link and it's underlying session and also removes it from the
+   * internal map.
+   * @ignore
+   * @param {Sender | Receiver} [link] The Sender or Receiver link that needs to be removed.
+   * @returns {void} void
+   */
+  protected _removeLink(link?: Sender | Receiver): void {
+    clearTimeout(this._tokenRenewalTimer as NodeJS.Timer);
```

Comment:
@bterlson could this cast fail when we do the browserification of Event Hub?

Reply:
Yes. When we get to browserification, this will need to be updated.

@ShivangiReja Since `_closeLink` (which will clear this timeout) will be called for all senders/receivers that get detached, any reason to included it here as well?

Reply:
Yes, In our case it will clear the timeout inside `detach()`  but `_removeLink` and `_closeLink` are two separate methods so I was thinking if somebody wants to use `_removeLink(link?: Sender | Receiver)` method in future then it is good to have clear timeout inside remove link.